forte/data/readers/conll03_reader_new.py

```diff
@@ -0,0 +1,149 @@
+# Copyright 2019 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+The reader that reads CoNLL ner_data into our internal json data datasets.
+"""
+import codecs
+import logging
+import os
+from typing import Iterator, Any
+
+from forte.data.data_pack import DataPack
+from forte.data.data_utils_io import dataset_path_iterator
+from forte.data.readers.base_reader import PackReader
+from ft.onto.base_ontology import Token, Sentence, Document, EntityMention
+
+__all__ = [
+    "CoNLL03Reader"
+]
+
+
+class CoNLL03Reader(PackReader):
+    r""":class:`CoNLL03Reader` is designed to read in the CoNLL03-ner dataset.
+    """
+
+    def _collect(self, conll_directory) -> Iterator[Any]:  # type: ignore
+        r"""Iterator over conll files in the data_source.
+
+        Args:
+            conll_directory: directory to the conll files.
+
+        Returns: Iterator over files in the path with conll extensions.
+        """
+        logging.info("Reading .conll from %s", conll_directory)
+        return dataset_path_iterator(conll_directory, "conll")
+
+    def _cache_key_function(self, conll_file: str) -> str:
+        return os.path.basename(conll_file)
+
+    def _parse_pack(self, file_path: str) -> Iterator[DataPack]:
+        pack = self.new_pack()
```

Comment:
Let's don't use the new_pack function now. Just create the pack directly `pack = DataPack()`